### PR TITLE
Upgrade Planner RCP Application

### PR DESCRIPTION
**Pull request by jtydhr88**

### CODE REVIEW

File: build/com.liferay.ide.upgrade.planner-repository/upgradePlanner.product [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?pde version="3.5"?>
+
+<product name="upgrade-planner" uid="upgradePlaner" id="com.liferay.ide.upgrade.planner.ui.upgradePlannerProduct" application="com.liferay.ide.upgrade.planner.ui.upgradePlannerApplication" version="3.6.0.qualifier" useFeatures="true" includeLaunchers="true">
+
+   <configIni use="default">
+   </configIni>
+
+   <launcherArgs>
```
[the remark this answers is not in the document]
Comment (jtydhr88) [reply build/com.liferay.ide.upgrade.planner-repository/upgradePlanner.product]: Right now I don't find any args needs in it.

File: build/parent/pom.xml [hunk 2]
```diff
@@ -47,6 +47,7 @@
         <swtbot-site>http://download.eclipse.org/technology/swtbot/releases/2.6.0/</swtbot-site>
         <sapphire-site>http://download.eclipse.org/sapphire/9.1.1/repository/</sapphire-site>
         <maven-deps-plugin-site>http://ianbrandt.github.io/m2e-maven-dependency-plugin/0.0.4/</maven-deps-plugin-site>
+        <emf-site>https://download.eclipse.org/modeling/emf/emf/builds/release/2.17</emf-site>
```
[the remark this answers is not in the document]
Comment (jtydhr88) [reply build/parent/pom.xml]: the rcp need emf but it doesn't in any repos we already have

File: upgrade-planner/plugins/com.liferay.ide.upgrade.planner.ui/plugin.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?eclipse version="3.4"?>
+<plugin>
+
+	<extension
+		point="org.eclipse.ui.perspectives"
+	>
+		<perspective
+			class="com.liferay.ide.upgrade.planner.ui.UpgradePlannerPerspective"
+			id="com.liferay.ide.upgrade.planner.perspective"
```
[the remark this answers is not in the document]
Comment (jtydhr88) [reply upgrade-planner/plugins/com.liferay.ide.upgrade.planner.ui/plugin.xml]: this one for rcp, and another persperctive for running in IDE

File: tools/plugins/com.liferay.ide.upgrade.plan.ui/src/com/liferay/ide/upgrade/plan/ui/util/Editors.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
+/**
```
[the remark this answers is not in the document]
Comment (jtydhr88) [reply tools/plugins/com.liferay.ide.upgrade.plan.ui/src/com/liferay/ide/upgrade/plan/ui/util/Editors.java]: the bundle com.liferay.ide.ui needs many bundles we don't need in planner, like jdt.ui, ant, and mylyn, we have to bundle them inside our rcp application if we depend on ide.ui

File: build/com.liferay.ide.upgrade.planner-repository/upgradePlanner.product (hunk 1 above)
[the remark this answers is not in the document]
Comment (gamerson) [reply build/com.liferay.ide.upgrade.planner-repository/upgradePlanner.product]: Does the resulting planner application have any arguments at all?

**Pull request closed without merging**